src/vs/workbench/contrib/chat/browser/tools/textSearchTool.ts

```diff
@@ -91,17 +101,22 @@ export class TextSearchTool implements IToolImpl {
 		}
 
 		// Set up the text search query
-		const patternInfo = invocation.parameters as TextSearchToolParams;
+		const searchParams = invocation.parameters as TextSearchToolParams;
 		const workspaceUris = workspaceFolders.map(folder => folder.uri);
+		// Don't allow more than the default max results, even if a higher value is provided,
+		// to prevent excessive token usage and performance issues.
+		const maxResults = searchParams.maxResults && searchParams.maxResults < DEFAULT_MAX_RESULTS
+			? searchParams.maxResults
+			: DEFAULT_MAX_RESULTS;
```

Comment:
Thoughts on capping the results limit? My thinking is that there just won't be a good experience if we allow more than probably 50-100 results, so capping seems like a reasonable safeguard.

We could specify different values for the default and the maximum if different numbers make sense for these!